ITensorGaussianMPS/src/gmps.jl

```diff
@@ -80,7 +132,11 @@ end
 
 function shift!(G::Circuit, i::Int)
   for (n, g) in enumerate(G.rotations)
-    G.rotations[n] = Givens(g.i1 + i, g.i2 + i, g.c, g.s)
+    if typeof(g) <: Givens
```

Comment:
I think it would be better to make use of a generic function `shift(g, i)` here, defined as:
```julia
shift(g::Givens, i::Int) = Givens(g.i1 + i, g.i2 + i, g.c, g.s)
shift(g::DiagRotation, i::Int) = DiagRotation(g.i1 + i, g.i2 + i, g.d)
```